packages/terra-form-select/src/Select.jsx

```diff
@@ -0,0 +1,297 @@
+import SelectMenu from './_SelectMenu';
+import SelectUtils from './_SelectUtils';
+import styles from './Select.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Function to trigger when the user changes the select value.
+   */
+  onChange: PropTypes.func,
+
+  /**
+   * Name of the select field.
+   */
+  name: PropTypes.string,
+
+  /**
+   * Whether the select is required or not.
+   */
+  required: PropTypes.bool,
+
+  /**
+   * The value to start the select on.
+   */
+  defaultValue: PropTypes.string,
+
+  /**
+   * The value of the select element. Use this to create a controlled input.
+   */
+  value: PropTypes.string,
+
+  /**
+   * Select options
+   */
+  children: PropTypes.node.isRequired,
+
+  /**
+   * Indicates if the select is in an invalid state.
+   */
+  isInvalid: PropTypes.bool,
+
+  /**
+   * Indicates if the select should be disabled.
+   */
+  disabled: PropTypes.bool,
+
+  /**
+   * A callback function to let the containing component (e.g. modal) to regain focus.
+   */
+  releaseFocus: PropTypes.func,
+
+  /**
+   * A callback function to request focus from the containing component (e.g. modal).
+   */
+  requestFocus: PropTypes.func,
+};
+
+const defaultProps = {
+  required: false,
+  disabled: false,
+  isInvalid: false,
+};
+
+const contextTypes = {
+  /* eslint-disable consistent-return */
+  intl: (context) => {
+    if (context.intl === undefined) {
+      return new Error('Please add locale prop to Base component to load translations');
+    }
+  },
+};
+
+class Select extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+    this.getInitialState = this.getInitialState.bind(this);
+    this.setTargetRef = this.setTargetRef.bind(this);
+    this.getTargetRef = this.getTargetRef.bind(this);
+    this.handleOnClick = this.handleOnClick.bind(this);
+    this.handleOnKeyDown = this.handleOnKeyDown.bind(this);
+    this.handleOnRequestClose = this.handleOnRequestClose.bind(this);
+    this.handleSelection = this.handleSelection.bind(this);
+    this.wrapOnClick = this.wrapOnClick.bind(this);
+    this.wrapOnKeyDown = this.wrapOnKeyDown.bind(this);
+    this.state = {
+      selectedValue: this.getInitialState(),
+      isOpen: false,
+    };
+  }
+
+  getInitialState() {
+    if (!this.props.value) {
+      const childArray = React.Children.toArray(this.props.children);
+      for (let i = 0; i < childArray.length; i += 1) {
+        const child = childArray[i];
+        if (child.props.isSelected || child.props.value === this.props.defaultValue) {
+          return child.props.value;
+        }
+      }
+    }
+    return '';
+  }
+
+  setTargetRef(node) {
+    this.targetRef = node;
+  }
+
+  getTargetRef() {
+    return this.targetRef;
+  }
+
+  wrapOnClick(option) {
+    return (event) => {
+      this.handleSelection(event, option);
+
+      if (option.props.onClick) {
+        option.props.onClick(event);
```

Comment:
It seems like returning the option in the onClick would be valuable.

Reply:
I'm not sure if I want to be passing helper info on the onClick since I would rather consumers didn't use the onClick at all, they _should_ be using onChange instead. That's why I didn't even expose onClick as a first class prop on the Option component.

Reply:
Okay I see what you mean. 

If we'd prefer consumers use OnChange and use the information returned their to perform other logic, should we even honor the onClick & onKeyDown they provided? I could see people expecting the same information the onChange would provide. 

Either way doesn't matter, just seems a bit odd to add the logic if the props aren't available & wrap our own functions for these events.

Reply:
I could be convinced either way, so I'll see if anyone else has opinions.